db-client-java/src/main/java/com/eventstore/dbclient/EventStoreDBPersistentSubscriptionsClient.java

```diff
@@ -164,4 +176,243 @@ public CompletableFuture deleteToAll(String group, DeletePersistentSubscriptionO
+                }
+                return ps;
+            } catch (IOException e) {
+                throw new RuntimeException(e);
+            } finally {
+                http.disconnect();
+            }
+        });
+    }
+
+    public CompletableFuture<List<PersistentSubscriptionInfo>> listAll() {
+        return listAll(ListPersistentSubscriptionsOptions.get());
+    }
+
+    public CompletableFuture<List<PersistentSubscriptionInfo>> listForStream(String stream, ListPersistentSubscriptionsOptions options) {
+        return getHttpURLConnection(options, "/subscriptions/" + stream).thenApply(http -> {
+            try {
+                http.setRequestMethod("GET");
+
+                throwOnError(http.getResponseCode());
+
+                String content = readContent(http);
+                List<PersistentSubscriptionInfo> ps = new ArrayList<>();
+
+                for (JsonNode node: mapper.readTree(content)) {
+                    ps.add(parseSubscriptionInfo(node));
+                }
+                return ps;
+            } catch (IOException e) {
+                throw new RuntimeException(e);
+            } finally {
+                http.disconnect();
+            }
+        });
+    }
+
+    public CompletableFuture<List<PersistentSubscriptionInfo>> listForStream(String stream) {
+        return listForStream(stream, ListPersistentSubscriptionsOptions.get());
+    }
+
+    public CompletableFuture<Optional<PersistentSubscriptionInfo>> getInfo(String stream, String groupName, GetPersistentSubscriptionInfoOptions options) {
+        return getHttpURLConnection(options, "/subscriptions/" + stream + "/" + groupName + "/info").thenApply(http -> {
+            try {
+                http.setRequestMethod("GET");
+                int code = http.getResponseCode();
+
+                if (code == 404)
+                    return Optional.empty();
+
+                throwOnError(code);
+
+                String content = readContent(http);
+                JsonNode node = mapper.readTree(content);
+
+                return Optional.of(parseSubscriptionInfo(node));
+            } catch (IOException e) {
+                throw new RuntimeException(e);
+            } finally {
+                http.disconnect();
+            }
+        });
+    }
+
+    public CompletableFuture<Optional<PersistentSubscriptionInfo>> getInfo(String stream, String groupName) {
+        return getInfo(stream, groupName, GetPersistentSubscriptionInfoOptions.get());
+    }
+
+    public CompletableFuture replayParkedMessages(String stream, String groupName, ReplayParkedMessagesOptions options) {
+        String query;
+
+        if (options.getStopAt() != null) {
+            query = "?stopAt=" + options.getStopAt().getSeconds();
+        } else {
+            query = "";
+        }
+
+        return getHttpURLConnection(options, "/subscriptions/" + stream + "/" + groupName + "/replayParked" + query).thenApply(http -> {

```

Comment:
**FIX:** I think we can do it right now through gRPC since when this PR was merged: https://github.com/EventStore/EventStore/pull/3352.